tests/integration/pilot/gateway_test.go

```diff
@@ -149,10 +149,17 @@ spec:
 }
 
 func ManagedGatewayTest(t framework.TestContext) {
-	t.ConfigIstio().YAML(apps.Namespace.Name(), `apiVersion: gateway.networking.k8s.io/v1beta1
+	templateArgs := map[string]string{
+		"revision": t.Settings().Revision,
+	}
+	t.ConfigIstio().Eval(apps.Namespace.Name(), templateArgs, `apiVersion: gateway.networking.k8s.io/v1beta1
 kind: Gateway
 metadata:
   name: gateway
+  {{- if .revision }}
```

Comment:
If the revision is the default revision, we shouldn't need to label the gateway right? It would get created by whatever revision is the default revision

Reply:
If the tests are ran using an specific istio revision, which is not the default, the gateway will not be programmed unless we explicitly set this label

Reply:
I understand your concern.

I ran the tests again without setting the revision label in all the tests (TestManagedGateway, TestUnmanagedGateway) and:

- In `TestManagedGateway` is not necessary as we install the gateway in the test namespace, which already has the `istio.io/rev` label
- In `TestUnmanagedGateway`, the Gateway is installed in `istio-system` (by default, which is where the ingress gateway is installed by default), and that namespace does not have `istio.io/rev` label. Thus, we still need the tag in here to make sure the test passes if a different namespace is provided